addons/mrp/models/mrp_production.py

```diff
@@ -792,7 +792,7 @@ def write(self, vals):
             if any(field in ['move_raw_ids', 'move_finished_ids', 'workorder_ids'] for field in vals) and production.state != 'draft':
                 production._autoconfirm_production()
                 if production in production_to_replan:
-                    production._plan_workorders()
+                    production._plan_workorders(replan=True)
```

Comment:
I don't remember what was the reason for @ajf-odoo removed this `replan=True` in the first place. I think it's OK to add back in, but maybe he should double check.

Reply:
I think its better to change in mrp.workorder._plan_workorder() the lines
`    if self.date_planned_start:
        if replan:
            self.leave_id.unlink()
        else:
            return
`for
    if self.leave_id:
        ...
as self.date_planned_start was used to consider the workorder as planned
but it is really planned when a leave has been allocated

Reply:
Using the case of Tiffany : https://drive.google.com/file/d/1EflKAnXtFSCSX9C6jCq5q-99QmaGoSDN/view?usp=drivesdk ,
After The 2nd WO is saved and the MO is confirmed :
![image](https://user-images.githubusercontent.com/109517847/197989333-141abbe6-319c-4091-b7f0-c8a7ca4d18c6.png)
I schedule the 1st WO (OP1) and save manually, with a breakpoint in `mrp_production._plan_workorders` : 
![replan false leave_id](https://user-images.githubusercontent.com/109517847/197989977-b905e336-acf1-47ab-b074-05de7d76e0ec.png)
We can see that both WO have a leave_id, despite the fact that OP1 is not yet planned as we just entered the method...
However, `OP2.leave_id` as no `date_from` assigned, although it's already planned.

Replacing the condition would so make every call to enter the condition, where we just want to enter it on replanning, which is nicely done (and moreover explicit, so easy to read) by setting `replan=True` as parameter in `mrp_production.write` method.

![write top](https://user-images.githubusercontent.com/109517847/197992884-bc7d7d47-ed20-4971-b3aa-0e9168384ea9.png)
![write bottom](https://user-images.githubusercontent.com/109517847/197992915-2fa437ec-75b8-42e7-8d4e-755cbd3f3c88.png)

As we can see with the 2 previous screenshots, the plan_workorders(replan=true) call is only done if the mo was already planned.

Reply:
It solves the problem :+1: